pepper-apis/src/main/java/org/broadinstitute/ddp/cache/CacheService.java

```diff
@@ -0,0 +1,153 @@
+package org.broadinstitute.ddp.cache;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.expiry.CreatedExpiryPolicy;
+import javax.cache.expiry.Duration;
+
+import org.broadinstitute.ddp.exception.DDPException;
+import org.jdbi.v3.core.Handle;
+
+
+public class CacheService {
+    private static CacheService instance;
+    private CacheManager cacheManager;
+    private Map<ModelChangeType, Collection<String>> modelChangeTypeToCacheName = new ConcurrentHashMap<>();
+    private Map<String, IdToCacheKeyMapper<?>> cacheNameToCacheKeyMapper = new ConcurrentHashMap<>();
+    private Map<String, IdToCacheKeyCollectionMapper<?>> cacheNameToCacheKeyCollectionMapper = new ConcurrentHashMap<>();
+
+    public static CacheService getInstance() {
+        if (instance != null) {
+            return instance;
+        } else {
+            synchronized (CacheService.class) {
+                if (instance == null) {
+                    instance = new CacheService();
+                }
+            }
+            return instance;
+        }
+    }
+
+    private CacheService() {
+        cacheManager = buildCacheManager();
+    }
+
+    private CacheManager buildCacheManager() {
+        URL resourceUrl = CacheService.class.getResource("/redisson-jcache.yaml");
+
+        URI redissonConfigUri;
+        try {
+            redissonConfigUri = resourceUrl.toURI();
+        } catch (URISyntaxException e) {
+            throw new DDPException(e);
+        }
+
+        return Caching.getCachingProvider().getCacheManager(redissonConfigUri, null);
+    }
+
+    public <K, V> Cache<K, V> getOrCreateCache(String cacheName, Duration entryDuration, IdToCacheKeyMapper<K> mapper,
+                                               ModelChangeType evictionModelChangeType, Object cacheParentObject) {
+        return _getOrCreateCache(cacheName, entryDuration, mapper, evictionModelChangeType, cacheParentObject);
+    }
+
+    public <K, V> Cache<K, V> getOrCreateCache(String cacheName, Duration entryDuration, IdToCacheKeyCollectionMapper<K> mapper,
+                                               ModelChangeType evictionModelChangeType, Object cacheParentObject) {
+        return _getOrCreateCache(cacheName, entryDuration, mapper, evictionModelChangeType, cacheParentObject);
+    }
```

Comment:
In addition to get and creating caches, also want to centralize management of cache invalidation. That what the key mappers and evictionModelChangeType stuff is about. More below